src/util.js

```diff
@@ -66,3 +66,60 @@ exports.handleServerError = (ctx, operationFailureMsg, error, logger) => {
   ctx.body = {error: err}
   logger.error(err)
 }
+
+// Strip away opening and closing forward slash if they are present in path
+const removeEnclosingSlashes = path => {
+  path = path.trim()
+
+  if (path[0] === '/') {
+    path = path.substring(1)
+  }
+  if (path[path.length - 1] === '/') {
+    path = path.substring(0, path.length - 1)
+  }
+
+  return path
+}
+
+exports.extractRegexFromPattern = pattern => {
+  if (pattern[0] === '/') {
+    pattern = pattern.substring(1)
+  }
+
+  const splitPattern = pattern.split(/\//g)
+  let regexString = ''
+  const urlParamRegexPart = new RegExp(/\/[^ ;:=#@,/]{1,}/)
```

Comment:
I see this would affect how duplicates are detected because we use the parameter names. We could get around that by instead of storing an extra property `patternRegex` in mongo, just searching for a similar `pattern` in mongo on saving an endpoint. E.g. replace ever url part starting with `:` with `/[^ ;:=#@,/]{1,}/` then using mongo regex find: https://docs.mongodb.com/manual/reference/operator/query/regex/ It also then gets rid of the risk of `pattern` and `patternRegex` getting out of sync.

I know this changes a bit, bit I think it is worth it as the code will be much simpler and it reduces some of the risk in this complex bit of code.

Reply:
hey man addressed in https://github.com/jembi/openhim-mediator-mapping/pull/97/commits/d082d1b1d3fa6ddcdda8e09bf27d42dbd09f8851